### PR TITLE
build(multiview-container + dependencies): use node16 module resolution

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to node16 module resolution.

Includes removal of unimplemented test stubs and unused test infrastructure.

### CODE REVIEW

File: examples/data-objects/multiview/container/package.json [hunk 1]
```diff
@@ -11,14 +11,12 @@
 	},
 	"license": "MIT",
 	"author": "Microsoft and contributors",
-	"main": "lib/index.js",
-	"module": "lib/index.js",
-	"types": "lib/index.d.ts",
+	"type": "module",
 	"scripts": {
```
Comment (jason-ha) [inline examples/data-objects/multiview/container/package.json:15]: This one looks like it is meant to be library+app and thus should have exports. I think the indicator is src/index.ts with something like `export const fluidExport`. @ChumpChief may know more.

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: examples/data-objects/multiview/container/package.json (hunk 1 above)
Comment (jason-ha) [reply examples/data-objects/multiview/container/package.json:15]: Update: pattern has changed but stale things like these package entrypoints weren't updated. New pattern is to use webpacked result. (Not sure what to check for to make sure webpack'ed library is sound. But that can be another day's concern and at least multiview-container is private package.)

File: examples/data-objects/multiview/container/tsconfig.json [hunk 2]
```diff
@@ -1,9 +1,7 @@
 {
-	"extends": "@fluidframework/build-common/ts-common-config.json",
+	"extends": "../../../../common/build/build-common/tsconfig.node16.json",
 	"compilerOptions": {
-		"module": "esnext",
-		"outDir": "lib",
-		"jsx": "react",
+		"outDir": "./lib",
```
Comment (jason-ha) [inline examples/data-objects/multiview/container/tsconfig.json:4]: retains another lib + test tsconfig that should be separated.

**Review by jason-ha — APPROVED** [review]
I wondered if the stub tests were actually providing value - just enough to check that the component gets loaded. But I can break them (make them throw) and test happily passes.
Additionally, I tried starting constellation-view. But it does not load.

There is no fluidExport let alone with an IRuntimeFactory. So, nothing loads in current state. stack:
```
container.ts:2389  Uncaught (in promise) Error: Code package does not implement IRuntimeFactory
    at Container.instantiateRuntime (container.ts:2389:10)
    at async Container.createDetached (container.ts:1797:3)
    at async _fluidframework_telemetry_utils__WEBPACK_IMPORTED_MODULE_6__.PerformanceEvent.timedExecAsync.start (container.ts:467:5)
    at async PerformanceEvent.timedExecAsync (logger.ts:725:16)
    at async Module.start (loader.ts:268:16)
```

**Comment by ChumpChief** [comment]
multiview-container is the only one that really wants to retain its test infra, seems correct to remove the rest (it uses all the rest, so it can serve as a sufficient smoke test for everything at once).

multiview-container is intentionally the only one with an export and able to be run directly (description of the demo structure here: https://github.com/microsoft/FluidFramework/tree/main/examples/data-objects/multiview#readme).  This demo dates back to when views shipped as integrated render() methods of data objects, so I went a little overboard with separation to overemphasize the point that the view can/should be separate.  Now that we've laid that issue to rest, we could probably consider whether there's a more concise way to demonstrate the multiview patterns, maybe a little less package-heavy.

**Comment by jason-ha** [comment]
Sounds like the starts should be removed from most of the packages too, then. Totally fine with that happening secondarily.

**Pull request merged**